### ISSUE TITLE
Confused about BCL

### ISSUE THREAD

**Issue by LeavesLi1015** [liubin06/BNS#1]
Hello! I'm really interested in your work and I have a little question about the `BCL` in your code.
In your paper, the prior distribution is caculated in one patch, 
<img width="262" alt="image" src="https://user-images.githubusercontent.com/76167689/224749984-88425657-e525-419a-879f-30504e4f83dc.png">
but in your code it writes `X = torch.cat([scores, scores[idx - 1, :],scores[idx - 2, :],scores[idx - 3, :]], dim=1).unsqueeze(1)` which means 4 patches needed. Why is there a change?

**Comment by liubin06** [comment]
> Hello! I'm really interested in your work and I have a little question about the `BCL` in your code. In your paper, the prior distribution is caculated in one patch, <img alt="image" width="262" src="https://user-images.githubusercontent.com/76167689/224749984-88425657-e525-419a-879f-30504e4f83dc.png"> but in your code it writes `X = torch.cat([scores, scores[idx - 1, :],scores[idx - 2, :],scores[idx - 3, :]], dim=1).unsqueeze(1)` which means 4 patches needed. Why is there a change?

Regarding your question about the BCL in my code, you are correct that the implementation involves the additional 3 patches, resulting in 4*N samples instead of the N samples mentioned in the paper, for the calculation of the empirical C.D.F. This change was made because the empirical C.D.F estimator converges to the C.D.F as N → ∞, with the standard $\sqrt {N}$ rate of convergence. A larger N yields higher accuracy for estimating the C.D.F. For a more detailed discussion, please refer to the following reference: [1] Van der Vaart, A. W. (2000). Asymptotic statistics (Vol. 3). Cambridge University Press.

While this trick is necessary when N is relatively small, it should be noted that this implementation of the code does not necessarily imply superiority since the assumption of identical distribution for the 4N samples may be too strong. Therefore, we have amended the code to provide both options for calculating the empirical distribution function, and users can choose the one that suits their scenario.

If you find this reply helpful, please STAR this project.

**Comment by LeavesLi1015** [comment]
Thanks a lot!
